### DISCUSSION TITLE
Parse datetimes with timezones

### DISCUSSION THREAD

**Discussion by egeres** [fastapi/typer#878] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorials in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import datetime
from typing import Optional

import typer
from rich import print

def main(now: Optional[datetime.datetime] = typer.Option(None)):
    print(f"now: {now}")

if __name__ == "__main__":
    typer.run(main)
```

### Description

So the previous code runs easily with:
```shell
python foo.py --now 2024-07-06T01:05:00
```

But it crashes if I add timezone information
```shell
python foo.py --now 2024-07-06T01:05:00+05:00
```

The error I'm given is: `Invalid value for '--now': '2024-07-06T01:05:00+02:00' does not match the formats '%Y-%m-%d', '%Y-%m-%dT%H:%M:%S', '%Y-%m-%d %H:%M:%S'.` My question is: **How could I allow users to input timezone information?**

### Operating System

Windows

### Typer Version

0.12.3

### Python Version

Python 3.9.7

**Comment by svlandeg** [comment, marked as answer]
Hi!

You can find the answer to your question in the documentation here: https://typer.tiangolo.com/tutorial/parameter-types/datetime/#custom-date-format

In short, Typer allows you to specify the date formats that should be used instead of relying on the default ones.

You can edit your code to something like this:
```
def main(now: Optional[datetime.datetime] = typer.Option(..., formats=["%Y-%m-%dT%H:%M:%S%z"])):
    ...
```

With `%z` referring to the timezone offset (see [here](https://docs.python.org/3/library/datetime.html)) and then it should be able to parse your input string.

**Reply by egeres** [reply]
Thank you!!!!
